File: leetcode/go/length_of_last_word_test.go
package main

import (
	"testing"
)

// type test struct {
// 	s string
// 	o int
// }

// var tests = []test{
// 	{"Hello World", 5},
// 	{"   fly me   to   the moon  ", 4},
// 	{"luffy is still joyboy", 6},
// }

func TestLengthOfLastWord(t *testing.T) {
	var tests = []struct {
		s string
		o int
	}{
		{"Hello World", 5},
		{"   fly me   to   the moon  ", 4},
		{"luffy is still joyboy", 6},
	}

	for _, tst := range tests {
		if r := LengthOfLastWord(tst.s); r != tst.o {
			t.Errorf("Failure: %v != %v", r, tst.o)
		}
	}
}

File: leetcode/clojure/test/length_of_last_word_test.clj
(ns length-of-last-word-test
  (:require [clojure.test :refer [deftest testing is]]
            [length-of-last-word :refer [length-of-last-word]]))

(deftest length-of-last-word-test
  (testing "basic"
    (is (= (length-of-last-word "Hello World") 5))
    (is (= (length-of-last-word "   fly me   to   the moon  ") 4))
    (is (= (length-of-last-word "luffy is still joyboy") 6))))
